src/melee/ft/chara/ftCommon/ftCo_0B3E.c

```diff
@@ -0,0 +1,9469 @@
+lbl_800B4564:
+/* 800B4564 000B1144  7C 60 07 74 */	extsb r0, r3
+/* 800B4568 000B1148  98 1D 00 04 */	stb r0, 4(r29)
+/* 800B456C 000B114C  48 00 00 90 */	b lbl_800B45FC
+entry lbl_800B4570
+/* 800B4570 000B1150  C0 1A 00 2C */	lfs f0, 0x2c(r26)
+/* 800B4574 000B1154  88 7C 00 00 */	lbz r3, 0(r28)
+/* 800B4578 000B1158  FC 00 F0 40 */	fcmpo cr0, f0, f30
+/* 800B457C 000B115C  88 9C 00 01 */	lbz r4, 1(r28)
+/* 800B4580 000B1160  7C 63 07 74 */	extsb r3, r3
+/* 800B4584 000B1164  7C 84 07 74 */	extsb r4, r4
+/* 800B4588 000B1168  3B 9C 00 02 */	addi r28, r28, 2
+/* 800B458C 000B116C  40 81 00 14 */	ble lbl_800B45A0
+/* 800B4590 000B1170  88 1D 00 04 */	lbz r0, 4(r29)
+/* 800B4594 000B1174  7C 00 07 74 */	extsb r0, r0
+/* 800B4598 000B1178  7C 60 1A 14 */	add r3, r0, r3
+/* 800B459C 000B117C  48 00 00 10 */	b lbl_800B45AC
+lbl_800B45A0:
+/* 800B45A0 000B1180  88 1D 00 04 */	lbz r0, 4(r29)
+/* 800B45A4 000B1184  7C 00 07 74 */	extsb r0, r0
+/* 800B45A8 000B1188  7C 63 00 50 */	subf r3, r3, r0
+lbl_800B45AC:
+/* 800B45AC 000B118C  7C 03 20 00 */	cmpw r3, r4
+/* 800B45B0 000B1190  40 81 00 0C */	ble lbl_800B45BC
+/* 800B45B4 000B1194  7C 83 23 78 */	mr r3, r4
+/* 800B45B8 000B1198  48 00 00 14 */	b lbl_800B45CC
+lbl_800B45BC:
+/* 800B45BC 000B119C  7C 04 00 D0 */	neg r0, r4
+/* 800B45C0 000B11A0  7C 03 00 00 */	cmpw r3, r0
+/* 800B45C4 000B11A4  40 80 00 08 */	bge lbl_800B45CC
+/* 800B45C8 000B11A8  7C 03 03 78 */	mr r3, r0
+lbl_800B45CC:
+/* 800B45CC 000B11AC  7C 60 07 74 */	extsb r0, r3
+/* 800B45D0 000B11B0  98 1D 00 04 */	stb r0, 4(r29)
+/* 800B45D4 000B11B4  48 00 00 28 */	b lbl_800B45FC
+entry lbl_800B45D8
+/* 800B45D8 000B11B8  38 00 00 00 */	li r0, 0
+/* 800B45DC 000B11BC  90 1D 04 4C */	stw r0, 0x44c(r29)
+/* 800B45E0 000B11C0  90 1D 04 50 */	stw r0, 0x450(r29)
+/* 800B45E4 000B11C4  48 00 00 28 */	b lbl_800B460C
+lbl_800B45E8:
+/* 800B45E8 000B11C8  3C 60 80 3C */	lis r3, ftCo_803C5DD4@ha
+/* 800B45EC 000B11CC  CB C2 8F E8 */	lfd f30, ftCo_804D89C8
+/* 800B45F0 000B11D0  CB E2 8F F0 */	lfd f31, ftCo_804D89D0
+/* 800B45F4 000B11D4  3B C3 5D D4 */	addi r30, r3, ftCo_803C5DD4@l
+/* 800B45F8 000B11D8  3F E0 43 30 */	lis r31, 0x4330
+entry lbl_800B45FC
+/* 800B45FC 000B11DC  80 1D 04 4C */	lwz r0, 0x44c(r29)
+/* 800B4600 000B11E0  28 00 00 00 */	cmplwi r0, 0
+/* 800B4604 000B11E4  41 82 F8 B8 */	beq lbl_800B3EBC
+/* 800B4608 000B11E8  93 9D 04 50 */	stw r28, 0x450(r29)
+lbl_800B460C:
+/* 800B460C 000B11EC  BB 01 00 30 */	lmw r24, 0x30(r1)
+/* 800B4610 000B11F0  80 01 00 6C */	lwz r0, 0x6c(r1)
+/* 800B4614 000B11F4  CB E1 00 60 */	lfd f31, 0x60(r1)
+/* 800B4618 000B11F8  CB C1 00 58 */	lfd f30, 0x58(r1)
+/* 800B461C 000B11FC  CB A1 00 50 */	lfd f29, 0x50(r1)
+/* 800B4620 000B1200  38 21 00 68 */	addi r1, r1, 0x68
+/* 800B4624 000B1204  7C 08 03 A6 */	mtlr r0
+/* 800B4628 000B1208  4E 80 00 20 */	blr
+}
+#pragma peephole on
+#endif /* clang-format on */
+
+ASM UNK_RET ftCo_800B462C(UNK_PARAMS)
+#if !defined(MUST_MATCH) || defined(WIP)
```

Comment:
Interaction of MUST_MATCH and WIP here smells weird to me. Shouldn't defining MUST_MATCH always result in matching code regardless of WIP state?

Reply:
No, because of things like `todo unused stack`. Also, if `MWERKS_GEKKO && !MUST_MATCH && !WIP` then the original asm should be used because that build is meant to be non-matching but as functional as possible, whereas a WIP function could be barely finished. `WIP` also causes the Makefile to build to a separate directory so it can be used with objdiff/asm-differ without rebuilding the whole project.

I wouldn't say it's a perfect system, but it offers a pretty smooth workflow with minimal setup.

Welcome back, btw.